### ISSUE TITLE
Discussion on functionality of moja-global contributors website

### ISSUE THREAD

**Issue by waridrox** [moja-global/GSOD.Contributors_website#8]
**Describe the bug**
The current contributor website can be found [here](https://contributors-website.herokuapp.com). 
The following components of the website were found to be non-functional: 
- Unresponsive navigation buttons - 
![image](https://user-images.githubusercontent.com/58583793/104778543-87872980-57a3-11eb-955e-769f957754a1.png)

- Very long loading time.
![image](https://user-images.githubusercontent.com/58583793/104778661-be5d3f80-57a3-11eb-888f-37b790a3c232.png)

- The beginner-friendly issue [list](https://contributors-website.herokuapp.com/#joinUsComponent:~:text=Beginner%20Friendly%20Issue%20List) is not updating according to the GitHub `label: good first issue`.

**To Reproduce**
1. Go to 'https://contributors-website.herokuapp.com/'
2. Wait for a few seconds to load the website (slow loading time). Then try to click on the specific buttons as described above.
3. Scroll down to the beginner-friendly issue list and check for different issues available in different repositories. 
4. See the errors. 

**Expected behavior**
Buttons should point to their respective references.
Loading time should be reduced as the users would assume the site to be unresponsive / down. 
Beginner-friendly issue list should link to issues targetted with the `label: good first issue` in their respective repositories. 

**Additional context**
I would like to work on the specified issues as described above.  

CC: @gmajan @aornugent  @Patamap

**Comment by aornugent** [comment]
Hey @waridrox - I moved this issue. Please feel free to work on the problems you have outlined. If the addressing the loading time requires any substantial changes to the site architecture please collaborate with the other contributors @Tlazypanda @shubhamkarande13 @HarshCasper to discuss the best approach.

**Comment by Tlazypanda** [comment]
@waridrox Hey there! Actually addressing some of the concerns here :sweat_smile: : - 

- The buttons haven't been assigned any link so far after discussion with @gmajan since we are yet to move the moja global website.
- The label issue is also there because we are in the process of migrating to proper labels like Github follows the good first issues approach but we have Good first issue as one of our organization labels.
- The slow loading time I believe is more due to we are using a free heroku setup which idles off if activity is shut down for 30 mins instead of being up the entire time.

**Comment by waridrox** [comment]
@Tlazypanda Ok, great! I thought the image loading time was a result of uncompressed images 😅. Didn't knew heroku cuts off the resource consumption. 

- Also if we could filter the selection of the repositories even further based upon if the repository contains at least one good first issue then only we display that and since the repositories are already filtered one step ahead it becomes even more easier for the newcomers to approach the issues. 

 - And if the number of good first issues reduces to zero that repository can be removed from the drop-down selection so that the users don't have to search in every repository to find good first issues. 
In other words, instead of displaying various repositories we only display those repositories that have `(good first issues >= 1)`

I would like to try and implement the above if it sounds reasonable...

**Comment by waridrox** [comment]
**Quick update on the functionality of the good first issue dropdown -**

- I discovered that the [GitHub API for issues](https://docs.github.com/en/rest/reference/issues) doesn’t allow for the good first issues to be filtered by an organisation (it only allows to search for issues with a label in a particular repository of that particular organisation). 
- Since the repository names are hardcoded, this may lead to contributors website with no good first issues, as all the good first issues may run out from a particular repository. Currently, this problem is resolved by manually updating the repository names every now and then.
- Thus, to overcome this problem I was working on a web scraper that can scrape the repository names with good first issues, run the script on the cloud every 24 hours (due to free resource constraints) and store the repo names output in a variable which can be fetched directly at the render time of the website. 

Hence, there is no need to change the repo names every now and then, plus the dropdown list is filtered one step ahead as the repo names are already sorted by good first issues. 

_Here is the implementation for the same in action -_ 

![Feb-03-2021 11-37-00](https://user-images.githubusercontent.com/58583793/106705616-4ea8da80-6614-11eb-84fd-9d0859c68d52.gif)


@Tlazypanda Please update the code structure so that I can implement the functionality asap 😇. I think this would be very helpful for new contributors.

Also I request @aornugent @gmajan and @Patamap to close the issues that have been resolved wherever possible like this issue [here](https://github.com/moja-global/FLINT/issues/66).

**Comment by Tlazypanda** [comment]
Hey @waridrox can we instead first instead fetch the repos from github api and then from this list apply our query on every repo to filter out the ones that don't have issues with the first timer labels? :sweat_smile: This way we don't have to maintain a scraper and cloud constraints.

**Comment by waridrox** [comment]
Sure @Tlazypanda, I actually thought about this but instead of collecting 39 repo names (and more if new repos are created) and running the query on every single one of them every time, wouldn't it be better to instead fetch the repo names from this[ link](https://github.com/issues?q=is%3Aissue+user%3Amoja-global+label%3A%22good+first+issue%22+is%3Aopen) directly and maintain a JSON output that updates every day and can be fetched as many times 😅? To make myself more clear, cloud constraints are applying only to the running of the scraper i.e. every day and not on the JSON output. It can be fetched as many times and that too very quickly.

**Comment by waridrox** [comment]
Sure @Tlazypanda, we can do this but instead of collecting 39 repo names (and more if new repos are created) and running the query on every single one of them every time, wouldn't it be better to instead fetch the repo names from this[ link](https://github.com/issues?q=is%3Aissue+user%3Amoja-global+label%3A%22good+first+issue%22+is%3Aopen) directly and maintain a JSON output that updates every day and can be fetched as many times 😅? To make myself more clear, cloud constraints are applying only to the running of the scraper i.e. every day and not on the JSON output. It can be fetched as many times and that too very quickly.

**Comment by waridrox** [comment]
Please give your opinions on this issue, @Tlazypanda @shubhamkarande13 and @HarshCasper, so we can implement the best approach possible.
